docs/source/reference/prototype.rst

```diff
@@ -21,47 +25,132 @@ item assignment and many more features.
+    device=None,
+    is_shared=False)
+
+
+:obj:`@tensorclass` supports indexing. Internally the tensor objects gets indexed,
+however the non-tensor data remains same
+
+.. code-block::
+
+
+ >>>print("indexed:", data[:2])
+ indexed: MyData(
+    floatdata=Tensor(shape=torch.Size([2, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+    intdata=Tensor(shape=torch.Size([2, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+    non_tensordata='test',
+    nested=MyData(
+        floatdata=Tensor(shape=torch.Size([2, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+        intdata=Tensor(shape=torch.Size([2, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+        non_tensordata='nested_test',
+        nested=None,
+        batch_size=torch.Size([2, 4]),
+        device=None,
+        is_shared=False),
+    batch_size=torch.Size([2, 4]),
+    device=None,
+    is_shared=False)
+
+:obj:`@tensorclass` also supports and get and set attributes. One can also access
+nested tensor class data
+
+.. code-block::
+
+ >>> data.non_tensordata = "test_changed"
+ >>> print("data.non_tensordata: ", repr(data.non_tensordata))
+ data.non_tensordata: 'test_changed'
+
+ >>> data.floatdata = torch.ones(3, 4, 5)
+ >>>print("data.floatdata:", data.floatdata)
+ data.floatdata: tensor([[[1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.]],
+
+        [[1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.]],
+
+        [[1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.]]])
 
+ >>> # Changing nested tensor data
+ >>> data.nested.non_tensordata = "nested_test_changed"
+ >>>print("data.nested.non_tensordata:", repr(data.nested.non_tensordata))
+ data.nested.non_tensordata: 'nested_test_changed'
+
+:obj:`@tensorclass` supports torch functions and the behavior is extened to
+nested tensor classes as well.Note that the operations are strictly restricted
+to tensor data types
+
+Here is an example:
+
+.. code-block::
 
+ >>>data2 = data.clone()
+ >>>cat_tc = torch.cat([data, data2], 0)
+ >>>print("Concatenated data:", catted_tc)
+ Concatenated data: MyData(
+    floatdata=Tensor(shape=torch.Size([6, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+    intdata=Tensor(shape=torch.Size([6, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+    non_tensordata='test_changed',
+    nested=MyData(
+        floatdata=Tensor(shape=torch.Size([6, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+        intdata=Tensor(shape=torch.Size([6, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+        non_tensordata='nested_test_changed',
+        nested=None,
+        batch_size=torch.Size([6, 4]),
+        device=None,
+        is_shared=False),
+    batch_size=torch.Size([6, 4]),
+    device=None,
+    is_shared=False)
```

Comment:
Let's create an "edge cases" section where we describe less intuitive behaviours, such as `__eq__`, what happens when a `None` value is assigned with a tensor and so forth.

Reply:
> Let's create an "edge cases" section where we describe less intuitive behaviours, such as `__eq__`, what happens when a `None` value is assigned with a tensor and so forth.

@vmoens  Adding the following to the edge cases:
1. Equality and inequality operations ( non-tensor data not validated here)
2. Set_item (Non-tensor data is validated at identity instead of equality)
3. cat/stack operations (No validation for non-tensor data in the list  and first tensor class object non-tensor data is considered while returning)
4. Pre-allocation. 

Quick question on pre-allocation. Do we want to tell users how we save and change them internally? I think since its internal we can just add it to the normal operations saying that tensorclass supports pre-allocation instead of an edge case. Wdyt?

Reply:
Added it to Edge cases for now